using-mobile-extensions/adobe-campaignclassic/adobe-campaignclass-c-api-reference.md

```diff
@@ -0,0 +1,261 @@
+
+The trackNotificationReceive API should be called after a push message has been received in the app. On iOS it takes a dictionary as a parameter and the dictionary should contain the message id and delivery id retrieved from the received push message. On Android, the API takes a map containing the same message id and delivery id received from the push message.
+
+{% tabs %}
+
+{% tab title="Android" %}
+
+#### Java
+
+#### Syntax
+
+```java
+public static void trackNotificationReceive(final Map<String, String> trackInfo)
+```
+
+#### Example
+
+```java
+public class MyFirebaseMessagingService extends FirebaseMessagingService {
+
+	private NotificationManager notifManager;
+	private Activity testActivity;
+
+	@Override
+	public void onMessageReceived(RemoteMessage remoteMessage) {
+
+		System.out.println("From: " + remoteMessage.getFrom());
+
+		// Check if message contains a data payload.
+		if (remoteMessage.getData().size() > 0) {
+			System.out.println("Message data payload: " + remoteMessage.getData());
+            // Build the additional parameter map
+            // then send the tracking info for message received
+            String broadlogId = remoteMessage.getMessageId();
+        	String deliveryId = remoteMessage.getFrom();
+            Map<String,String> trackInfo = new HashMap<>();
+        	trackInfo.put("broadlogid", broadlogId);
+        	trackInfo.put("deliveryid", deliveryId);
+        	CampaignClassic.trackNotificationReceieve(trackInfo);
+	}
+}
+```
+
+{% endtab %}
+
+{% tab title="iOS" %}
+
+#### Objective-C
+
+#### Syntax
+
+```objectivec
++ (void) trackNotificationReceive: (nonnull NSDictionary<NSString*, NSString*>*) trackInfo;
+```
+
+#### Example
+
+```objectivec
+// Invoked when a notification is delivered to a foreground application (TrackReceive)
+-(void)userNotificationCenter:(UNUserNotificationCenter *)center willPresentNotification:(UNNotification *)notification withCompletionHandler:(void (^)(UNNotificationPresentationOptions options))completionHandler{
+    NSLog(@"User Info : %@",notification.request.content.userInfo);
+    // call trackNotificationReceive API to send receive information
+[ACPCampaignClassic trackNotificationReceive:notification.request.content.userInfo];
+completionHandler(UNAuthorizationOptionSound | UNAuthorizationOptionAlert | UNAuthorizationOptionBadge);
+}
+```
+
+#### Swift
+
+```swift
+ACPCampaignClassic.trackNotificationReceive(trackInfo[String:String])
+```
+
+{% endtab %}
+{% endtabs %}
+
+### trackNotificationClick
+
+The trackNotificationClick API should be called after a push message has been opened by the user and clickedthrough it to launch the app. Similar to the trackNotificationReceive API, It takes a dictionary as a parameter and the dictionary should contain the message id and delivery id retrieved from the push message which was clickedthrough. On Android, the API takes a map containing the same message id and delivery id received from the push message clickthrough.
```

Comment:
- In the second sentence, lower case "It" to "it"

- in the second sentence, please add a comma between "parameter" and "and" so that the sentence reads as follows:
"...a dictionary as a parameter, and the dictionary should..."

- Please change all instances of "id" to "ID"